redisearch/client.py

```diff
@@ -19,81 +19,69 @@ class Field(object):
     TAG = 'TAG'
     SORTABLE = 'SORTABLE'
     NOINDEX = 'NOINDEX'
-    SEPARATOR = 'SEPARATOR'
-    PHONETIC = 'PHONETIC'
 
-    def __init__(self, name, *args):
+    def __init__(self, name, *args, sortable=False, no_index=False):
         self.name = name
-        self.args = args
+        self.args = list(args)
+
+        if sortable:
+            self.args.append(Field.SORTABLE)
+        if no_index:
+            self.args.append(Field.NOINDEX)
+
+        if no_index and not sortable:
+            raise ValueError('Non-Sortable non-Indexable fields are ignored')
 
     def redis_args(self):
-        return [self.name] + list(self.args)
+        return [self.name] + self.args
 
 
 class TextField(Field):
     """
     TextField is used to define a text field in a schema definition
     """
     NOSTEM = 'NOSTEM'
+    PHONETIC = 'PHONETIC'
 
-    def __init__(self, name, weight=1.0, sortable=False, no_stem=False,
-                 no_index=False, phonetic_matcher=None):
+    def __init__(self, name, weight=1.0, no_stem=False, phonetic_matcher=None, **kwargs):
         args = [Field.TEXT, Field.WEIGHT, weight]
         if no_stem:
             args.append(self.NOSTEM)
-        if sortable:
-            args.append(Field.SORTABLE)
-        if no_index:
-            args.append(self.NOINDEX)
         if phonetic_matcher and phonetic_matcher in ['dm:en', 'dm:fr', 'dm:pt', 'dm:es']:
             args.append(self.PHONETIC)
             args.append(phonetic_matcher)
 
-        if no_index and not sortable:
-            raise ValueError('Non-Sortable non-Indexable fields are ignored')
-        Field.__init__(self, name, *args)
+        Field.__init__(self, name, *args, **kwargs)
```

Comment:
Usually you put the call to the father as the first call in the constructor. Maybe we can call it first and create an add_arg function of the father that the child can use to add the extra arguments after? WDYT?

Reply:
why not just use `self.args.append` directly (instead of adding the `add_arg` function)?

Reply:
Also, the sortable and no_index flags (which being added in `Field.__init__`) should be at the end

Reply:
Exactly because of that, if you expose it as a method then it's an API and the father can decide to implement it anyway it wants. He can save it on a separate array and manage the arguments positions (for example)..